src/groupeddataframe/splitapplycombine.jl

```diff
@@ -414,10 +416,45 @@ julia> combine(df -> sum(df.c), gd) # Slower variant
 
 See [`by`](@ref) for more examples.
 """
-function combine(f::Any, gd::GroupedDataFrame)
+combine(f::Function, gd::GroupedDataFrame; keepkeys::Bool=true) =
+    combine_helper(f, gd, keepkeys=keepkeys)
+combine(gd::GroupedDataFrame, f::Function; keepkeys::Bool=true) =
+    combine_helper(f, gd, keepkeys=keepkeys)
+
+function DataFrames.combine(gd::GroupedDataFrame, cs::Pair...; keepkeys::Bool=true)
+    if any(x -> first(x) isa Tuple, cs)
+        Base.depwarn("passing a Tuple as column selector is deprecated, use " *
+                     "an AbstractVector instead", :combine)
+        cs = collect(cs)
+        for (i, v) in enumerate(cs)
+            if first(v) isa Tuple
+                cs[i] = collect(first(v)) => last(v)
+            end
+        end
+    end
+    cs_norm = [normalize_selection(index(parent(gd)), c) for c in cs]
+    inverse_cs_norm = [to => (from =>fun) for (from, (fun, to)) in cs_norm]
+    @assert !isempty(inverse_cs_norm)
+    nt_cs = (; inverse_cs_norm...)
+    return combine_helper(nt_cs, gd, keepkeys=keepkeys)
+end
+
+function combine(gd::GroupedDataFrame; f...)
+    if length(f) == 0
```

Comment:
Maybe make this a separate method, and move the other branch with the deprecation warning do deprecated.jl?

Reply:
If I understand you correctly it is not possible - both `combine(gd)` and `combine(gd, some_kwargs)` are now deprecated, but they share the same method (as method is identified only by positional arguments).

Reply:
Ah right I missed the `;`.